src/main/java/com/google/devtools/build/lib/bazel/repository/RepoDefinitionFunction.java

```diff
@@ -99,10 +104,23 @@ public SkyValue compute(SkyKey skyKey, Environment env)
     }
 
     // Step 1: Look for repositories defined by non-registry overrides.
-    Optional<RepoSpec> repoSpec = checkRepoFromNonRegistryOverrides(root, repositoryName);
-    if (repoSpec.isPresent()) {
+    String overriddenModuleName =
```

Comment:
can we retain the previous structure and put lines 107-121 into the `checkRepoFromNonRegistryOverrides` method?